app/src/main/java/com/bruno/aybar/composechallenges/common/AnimationStateHolder.kt

```diff
@@ -1,96 +0,0 @@
-package com.bruno.aybar.composechallenges.common
-
-import androidx.compose.animation.core.AnimationClockObservable
-import androidx.compose.animation.core.TransitionDefinition
-import androidx.compose.animation.core.TransitionState
-import androidx.compose.runtime.Composable
-import androidx.compose.runtime.getValue
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.setValue
-import androidx.compose.ui.platform.AmbientAnimationClock
-import androidx.compose.ui.platform.AnimationClockAmbient
-
-abstract class AnimationStateHolder<T>(initialState: T) {
-
-    /**
-     * Represents the current state of the animation. Its value won't change
-     * until the animation to a new state has been completed.
-     */
-    var current by mutableStateOf(initialState)
-        private set
-
-    /**
-     * Represents the state to which we are transitioning to. If there is no transition
-     * occurring at the moment, then [animateTo] and [current] will have the same value.
-     */
-    var animatingTo by mutableStateOf(initialState)
-        private set
```

Comment:
This is pretty much the same as `currentState` and `targetState` from the new [MutableTransitionState](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/animation/animation-core/src/commonMain/kotlin/androidx/compose/animation/core/Transition.kt;l=83?q=Transition.kt&ss=androidx%2Fplatform%2Fframeworks%2Fsupport:compose%2F), so that one can be used instead.